src/blockchain/utils.ts

```diff
@@ -137,7 +137,7 @@ export const getExpectedRate = async (
     amount
   )
   return isMinRate
-    ? Math.round(Number(expectedRate.toString()) * 0.98)
```

Comment:
@michaelcohen716 - This was the root cause of the issue where we were sending `number` instead of `BigNumber` to the contract method. Any reason to do `Math.round` here? I've skipped it for the time being as `BigNumber` utils doesn't provide such functionality.